bundles/org.openhab.binding.goecharger/src/main/java/org/openhab/binding/goecharger/internal/handler/GoEChargerHandler.java

```diff
@@ -0,0 +1,333 @@
+                        count++;
+                    }
+                    if (goeResponse.energy[6] > 0) { // amps P3
+                        count++;
+                    }
+                    return new DecimalType(count);
+                case TEMPERATURE:
+                    return new QuantityType<>(goeResponse.temperature, SIUnits.CELSIUS);
+                case SESSION_CHARGE_CONSUMPTION:
+                    return new QuantityType<>((Double) (goeResponse.sessionChargeConsumption / 360000d),
+                            SmartHomeUnits.KILOWATT_HOUR);
+                case SESSION_CHARGE_CONSUMPTION_LIMIT:
+                    return new QuantityType<>((Double) (goeResponse.sessionChargeConsumptionLimit / 10d),
+                            SmartHomeUnits.KILOWATT_HOUR);
+                case TOTAL_CONSUMPTION:
+                    return new QuantityType<>((Double) (goeResponse.totalChargeConsumption / 10d),
+                            SmartHomeUnits.KILOWATT_HOUR);
+                case FIRMWARE:
+                    return new StringType(goeResponse.firmware);
+                case VOLTAGE_L1:
+                    return new QuantityType<>(goeResponse.energy[0], SmartHomeUnits.VOLT);
+                case VOLTAGE_L2:
+                    return new QuantityType<>(goeResponse.energy[1], SmartHomeUnits.VOLT);
+                case VOLTAGE_L3:
+                    return new QuantityType<>(goeResponse.energy[2], SmartHomeUnits.VOLT);
+                case CURRENT_L1:
+                    return new QuantityType<>((Double) (goeResponse.energy[4] / 10d), SmartHomeUnits.AMPERE);
+                case CURRENT_L2:
+                    return new QuantityType<>((Double) (goeResponse.energy[5] / 10d), SmartHomeUnits.AMPERE);
+                case CURRENT_L3:
+                    return new QuantityType<>((Double) (goeResponse.energy[6] / 10d), SmartHomeUnits.AMPERE);
+                case POWER_L1:
+                    return new QuantityType<>((Double) (goeResponse.energy[7] / 10d),
+                            SmartHomeUnits.WATT);
+                case POWER_L2:
+                    return new QuantityType<>((Double) (goeResponse.energy[8] / 10d),
+                            SmartHomeUnits.WATT);
+                case POWER_L3:
+                    return new QuantityType<>((Double) (goeResponse.energy[9] / 10d),
+                            SmartHomeUnits.WATT);
+            }
+        }
+        return UnDefType.UNDEF;
+    }
+
+    @Override
+    @SuppressWarnings("unchecked")
+    public void handleCommand(ChannelUID channelUID, Command command) {
+        if (command instanceof RefreshType) {
+            refresh();
+            return;
+        }
+
+        String key = null;
+        String value = null;
+        switch (channelUID.getId()) {
+            case MAX_AMPERE:
+                key = "amp";
+                value = ((QuantityType<?>)command).intValue() + "";
+                break;
```

Comment:
Can you be sure the command here will be always of QuantityType type? I would split this switch into two `if` checks: `if (command instanceof QuantityType)` and `if (command instanceof OnOffType)`.

Reply:
max amps is always quantity type, yes. How could it be something differentif i check for the ID beforehand?

Reply:
The channel is of QuantityType, but I am talking about passed command. I believe you can send whatever command you want using rules. Also, when there is new command introduced, something similar to `REFRESH` command, you will get cast exception here.

Reply:
Ok i get it. But then it gets quite complicated, command can be either string, or number or whatever. If you send Text Command "6" or  Number Command 6 it could both work.

Reply:
No. Commands are strongly typed in openHAB. Checking for `instanceof QuantityType` is enough. Commands send from rules are correctly casted before.

Reply:
Ok so i do not have to do anything here? Will quickly check with a custom rule...

Reply:
```
java.lang.ClassCastException: org.eclipse.smarthome.core.library.types.DecimalType cannot be cast to org.eclipse.smarthome.core.library.types.QuantityType
```
So i guess i have the wrong type here anyway.

Reply:
However if i send a string in the rule instead of an int value it still says decimaltype in the exception, so i guess no need to check for different types.

Reply:
Changed type to decimaltype and tested a rule with string and number and both worked.

Reply:
This is what I do: check type before casting, because the value comes fron the outside and there is no guarantee it will be of proper type. It happened to my binding (satel) when `REFRESH` command had been introduced - the binding was crashing on this command because there wasn't type checking in one handler. 
But if you feel that this is not needed here, just leave as it is.